pyvrp/tests/test_Population.py

```diff
@@ -177,24 +178,43 @@ def test_select_returns_same_parents_if_no_other_option():
 # // TODO test more select() - diversity, feas/infeas pairs
 
 
-def test_restart_generates_min_pop_size_new_individuals():
-    """
-    Tests if restarting the population will generate ``min_pop_size`` new
-    individuals.
-    """
-    data = read("data/RC208.txt", "solomon", "dimacs")
+def test_same_initial_solutions():
+    data = read("data/E-n22-k4.txt", round_func="round")
     pm = PenaltyManager(data.vehicle_capacity)
     rng = XorShift128(seed=12)
 
-    params = PopulationParams()
-    pop = Population(data, pm, rng, broken_pairs_distance, params)
+    params = PopulationParams(min_pop_size=10)
+    init = make_random_solutions(params.min_pop_size, pm, rng, data)
+    pop = Population(init, bpd, params)
+
+    # Check that the initial population individuals have the same routes as the
+    # passed-in initial solutions. We check for equality here because the
+    # population creates new individuals.
+    for indiv in pop:
+        assert_(np.any(indiv == other for other in init))
 
-    old = {individual for individual in pop}
     pop.restart()
-    new = {individual for individual in pop}
 
-    assert_equal(len(pop), params.min_pop_size)
-    assert_equal(len(old & new), 0)  # no old pops survived the restart
+    # Check again for the restarted population.
+    for indiv in pop:
+        assert_(np.any(indiv == other for other in init))
```

Comment:
I don't really get why this needs to be changed. Would something like:
```python
init = set(...)
current = {individual for individual in pop}
assert_(len(current & init), params.min_pop_size)

pop.restart()

new = {individual for individual in pop}
assert_(len(new & init), params.min_pop_size)
```
not work?

Reply:
No idea why I changed this - I changed it again to what you suggested!